src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.index = Animal.alive.indexOf(this);
+      Animal.alive.splice(this.index, 1);
+    }
```

Comment:
[CODE PERFORMANCE] The `checkHealth` method uses `indexOf` and `splice` within a loop which can lead to performance issues. It is preferred to use the `filter` method to remove dead animals from the `Animal.alive` array. Consider refactoring this logic to avoid nesting loops and improve performance.

Reply:
> [CODE PERFORMANCE] The `checkHealth` method uses `indexOf` and `splice` within a loop which can lead to performance issues. It is preferred to use the `filter` method to remove dead animals from the `Animal.alive` array. Consider refactoring this logic to avoid nesting loops and improve performance.

Good point